### ISSUE TITLE
Convergence Warning During shap.values()

### ISSUE THREAD

**Issue by jlevy44** [shap/shap#379]
Hey there! I'm using the gradient explainer to interpret a 8000 sample x 200000 feature array. I split it into training and testing, and subsampled to get 200 train samples and 200 test samples. When I run .shap_values(), I've been getting these convergence warning, and shap is unable to complete. This warning is printed out every backprop iteration. Can you advise on how to get rid of this?:

/xxx/rc/home/w/xxx/.local/lib/python3.6/site-packages/sklearn/linear_model/least_angle.py:311: ConvergenceWarning:

Regressors in active set degenerate. Dropping a regressor, after 12 iterations, i.e. alpha=7.091e-02, with an active set of 12 regressors, and the smallest cholesky pivot element being 5.373e-08. Reduce max_iter or increase eps parameters.

/xxx/rc/home/w/xxx/.local/lib/python3.6/site-packages/sklearn/linear_model/least_angle.py:337: ConvergenceWarning:

Early stopping the lars path, as the residues are small and the current value of alpha is no longer well controlled. 13 iterations, alpha=7.423e-02, previous alpha=7.071e-02, with an active set of 12 regressors.

**Comment by jlevy44** [comment]
Shapley nsamples is 100, and I'm using the top 30 ranked_outputs

**Comment by jlevy44** [comment]
Ah. Actually was using kernel explainer due to a bug that I fixed. It would be helpful to know why I am getting this for the kernel explainer though

**Comment by slundberg** [comment]
Those warnings are coming from the linear regression step inside of KernelExplainer. My best guess is that because you have so many regressors the automatic choice of l1_reg leads to an ill-conditioned problem. You might need to manually increase the regularization (by setting `l1_reg`) at that point. That being said, if it's a linear model you would do far better to leverage that fact and use the LinearExplainer.
